### ISSUE TITLE
`core::fmt::rt::Opaque` not `Sync` caused future compile fail on nightly-2024-03-09

### ISSUE THREAD

**Issue by PureWhiteWu** [rust-lang/rust#122674]
### Code

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=33a2a5c86f6e3276da4da323ac28ed31

### Version with regression

searched nightlies: from nightly-2024-03-07 to nightly-2024-03-10
regressed nightly: nightly-2024-03-09
searched commit range: https://github.com/rust-lang/rust/compare/9c3ad802d9b9633d60d3a74668eb1be819212d34...46b180ec2452d388c5d9c14009442e2e0beb01d7
regressed commit: https://github.com/rust-lang/rust/commit/a655e648a9f94d74263108366b83e677af56e35d

<details>
<summary>bisected with <a href='https://github.com/rust-lang/cargo-bisect-rustc'>cargo-bisect-rustc</a> v0.6.8</summary>

Host triple: aarch64-apple-darwin
Reproduce with:
```bash
cargo bisect-rustc --start 2024-03-08 --end 2024-03-09
```
</details>

@rustbot modify labels: +regression-from-stable-to-nightly -regression-untriaged

**Comment by PureWhiteWu** [comment]
I suspect #121563 caused this regression, but I don't know how to use `cargo-bisect-rustc` to test this pr(I tried some times but all failed).

Do you have any ideas about this? @Jarcho

**Comment by Nilstrieb** [comment]
if you use the latest version of cargo-bisect-rustc it will bisect the roll-up for you.

**Comment by PureWhiteWu** [comment]
@Nilstrieb Hi, I'm using the latest version of `cargo-bisect-rustc`, but seems that it fails to bisect the roll-up.

Here's the complete log:

```
looking for regression commit between 2024-03-08 and 2024-03-09
fetching (via remote github) commits from max(9c3ad802d9b9633d60d3a74668eb1be819212d34, 2024-03-06) to 46b180ec2452d388c5d9c14009442e2e0beb01d7
ending github query because we found starting sha: 9c3ad802d9b9633d60d3a74668eb1be819212d34
get_commits_between returning commits, len: 10
  commit[0] 2024-03-07: Auto merge of #119199 - dpaoliello:arm64ec, r=petrochenkov
  commit[1] 2024-03-07: Auto merge of #122151 - GuillaumeGomez:rollup-hfxr9kv, r=GuillaumeGomez
  commit[2] 2024-03-08: Auto merge of #122048 - erikdesjardins:inbounds, r=oli-obk
  commit[3] 2024-03-08: Auto merge of #122059 - nyurik:with-as-const-str, r=cuviper
  commit[4] 2024-03-08: Auto merge of #120268 - DianQK:otherwise_is_last_variant_switchs, r=oli-obk
  commit[5] 2024-03-08: Auto merge of #122182 - matthiaskrgr:rollup-gzimi4c, r=matthiaskrgr
  commit[6] 2024-03-08: Auto merge of #122078 - gurry:121443-ice-layout-is-sized-alt, r=oli-obk
  commit[7] 2024-03-08: Auto merge of #121500 - oli-obk:track_errors12, r=petrochenkov
  commit[8] 2024-03-08: Auto merge of #122190 - matthiaskrgr:rollup-9ol4y30, r=matthiaskrgr
  commit[9] 2024-03-08: Auto merge of #122206 - matthiaskrgr:rollup-4txx9wx, r=matthiaskrgr
validated commits found, specifying toolchains

checking the start range to verify it passes
installing 9c3ad802d9b9633d60d3a74668eb1be819212d34
rust-std-nightly-aarch64-apple-darwin: 25.41 MB / 25.41 MB [============================================================================================================================================================] 100.00 % 16.06 MB/s testing...
RESULT: 9c3ad802d9b9633d60d3a74668eb1be819212d34, ===> No
uninstalling 9c3ad802d9b9633d60d3a74668eb1be819212d34

checking the end range to verify it does not pass
installing 46b180ec2452d388c5d9c14009442e2e0beb01d7
rust-std-nightly-aarch64-apple-darwin: 25.39 MB / 25.39 MB [============================================================================================================================================================] 100.00 % 17.43 MB/s testing...
RESULT: 46b180ec2452d388c5d9c14009442e2e0beb01d7, ===> Yes
uninstalling 46b180ec2452d388c5d9c14009442e2e0beb01d7

5 versions remaining to test after this (roughly 3 steps)
installing 14fbc3c00525b41a3a3ee2c90e9ab6fd3b05274f
rust-std-nightly-aarch64-apple-darwin: 25.44 MB / 25.44 MB [============================================================================================================================================================] 100.00 % 19.68 MB/s testing...
RESULT: 14fbc3c00525b41a3a3ee2c90e9ab6fd3b05274f, ===> No
uninstalling 14fbc3c00525b41a3a3ee2c90e9ab6fd3b05274f

3 versions remaining to test after this (roughly 2 steps)
installing 42825768b103c28b10ce0407749acb21d32abeec
rust-std-nightly-aarch64-apple-darwin: 25.42 MB / 25.42 MB [============================================================================================================================================================] 100.00 % 18.31 MB/s testing...
RESULT: 42825768b103c28b10ce0407749acb21d32abeec, ===> No
uninstalling 42825768b103c28b10ce0407749acb21d32abeec

2 versions remaining to test after this (roughly 2 steps)
installing 74acabe9b042ea8c42862ee29aca2a8b7d333644
rust-std-nightly-aarch64-apple-darwin: 25.39 MB / 25.39 MB [============================================================================================================================================================] 100.00 % 19.89 MB/s testing...
RESULT: 74acabe9b042ea8c42862ee29aca2a8b7d333644, ===> No
uninstalling 74acabe9b042ea8c42862ee29aca2a8b7d333644

1 versions remaining to test after this (roughly 1 steps)
installing a655e648a9f94d74263108366b83e677af56e35d
rust-std-nightly-aarch64-apple-darwin: 25.41 MB / 25.41 MB [============================================================================================================================================================] 100.00 % 15.75 MB/s testing...
RESULT: a655e648a9f94d74263108366b83e677af56e35d, ===> Yes
uninstalling a655e648a9f94d74263108366b83e677af56e35d

searched toolchains 9c3ad802d9b9633d60d3a74668eb1be819212d34 through 46b180ec2452d388c5d9c14009442e2e0beb01d7


********************************************************************************
Regression in a655e648a9f94d74263108366b83e677af56e35d
********************************************************************************

Attempting to search unrolled perf builds
Found commits ["3e513c4c", "2d1c3361", "d2049e2a", "079e6df4", "1f4c0e1b", "de1b1944", "fd318cd5", "2ea39767"]
installing 3e513c4c2c66ab0c2fd138e3a53e4873277964f1
uninstalling 3e513c4c2c66ab0c2fd138e3a53e4873277964f1
installing 2d1c3361fe0fff25d8d5a8f02f9f94d8aa56e566
uninstalling 2d1c3361fe0fff25d8d5a8f02f9f94d8aa56e566
installing d2049e2a53ae398bba7a7e3f5c9519a721d07f0f
uninstalling d2049e2a53ae398bba7a7e3f5c9519a721d07f0f
installing 079e6df43c77f89c3d6d71bf563be219286904b5
uninstalling 079e6df43c77f89c3d6d71bf563be219286904b5
installing 1f4c0e1bed6ba336e5ee19076246c53f9f33e6df
uninstalling 1f4c0e1bed6ba336e5ee19076246c53f9f33e6df
installing de1b1944b3846544b36a05ac21b4a9520bf938ec
uninstalling de1b1944b3846544b36a05ac21b4a9520bf938ec
installing fd318cd5b1d3b39f6412149d5dceb8099683aefe
uninstalling fd318cd5b1d3b39f6412149d5dceb8099683aefe
installing 2ea39767a9dce310c992f55b5bd7dfac6bf531dd
uninstalling 2ea39767a9dce310c992f55b5bd7dfac6bf531dd
ERROR: none of the toolchains satisfied the predicate
==================================================================================
= Please file this regression report on the rust-lang/rust GitHub repository     =
=        New issue: https://github.com/rust-lang/rust/issues/new                 =
=     Known issues: https://github.com/rust-lang/rust/issues                     =
= Copy and paste the text below into the issue report thread.  Thanks!           =
==================================================================================

searched nightlies: from nightly-2024-03-08 to nightly-2024-03-09
regressed nightly: nightly-2024-03-09
searched commit range: https://github.com/rust-lang/rust/compare/9c3ad802d9b9633d60d3a74668eb1be819212d34...46b180ec2452d388c5d9c14009442e2e0beb01d7
regressed commit: https://github.com/rust-lang/rust/commit/a655e648a9f94d74263108366b83e677af56e35d

<details>
<summary>bisected with <a href='https://github.com/rust-lang/cargo-bisect-rustc'>cargo-bisect-rustc</a> v0.6.8</summary>


Host triple: aarch64-apple-darwin
Reproduce with:
```bash
cargo bisect-rustc --start 2024-03-08 --end 2024-03-09
```
</details>
```

**Comment by lqd** [comment]
It does bisect to https://github.com/rust-lang/rust/pull/121563 for me.

**Comment by lqd** [comment]
The fix has landed on master. Reopening to track beta backport.

**Comment by PureWhiteWu** [comment]
@lqd Hi, 1.77.0 has been released, can this be backported to beta now?

**Comment by lqd** [comment]
I wasn’t at the weekly meeting and I don’t know for sure: https://github.com/rust-lang/rust/pull/122680 is not yet marked as accepted for backport (it may just be an oversight in labeling, I’ll try to find out). Once that is done though, the release team, I believe, will backport it.

**Comment by apiraino** [comment]
updated labels on #122680 (sorry for the delay)

**Comment by cuviper** [comment]
That was backported in #123172.
